edit/listing.go

```diff
@@ -343,3 +384,201 @@ func getListing(ed *Editor) *listing {
 		panic("unreachable")
 	}
 }
+
+type listingItem interface {
+	eval.Value
+	Display() ui.Styled
+	Content() string
+	FilterText() string
+}
+
+type listingOptions struct {
+	modeline   string
+	autoCommit bool
+	keepBottom bool
+	ignoreDup  bool
+	ignoreCase bool
+}
+
+type listingItemString struct {
+	eval.String
+}
+
+func (s *listingItemString) Content() string {
+	return s.String.String()
```

Comment:
Did you mean the representation of `eval.String.String()` could be different from its internal string? Since I don't see much saving from `s.String.String()` to `string(s.String)`.